### PR TITLE
docs: add decision record about refactoring the OpenAPI documentatiion process

### LINKED ISSUE #2163: Documentation: Publish OpenAPI Spec to Swaggerhub in CI
# Feature Request

In addition to publishing the `openapi.yaml` file in our code base we should give users a way to inspect the API definition visually, i.e. on a pretty website.

We'll thus publish the `openapi.yaml` file to SwaggerHub. An example of how this could look like can be found [here](https://app.swaggerhub.com/apis/beardyinc/edc-rest_api/0.0.1-SNAPSHOT). For this, I manually uploaded the file, but this can be automated through CI.

We should then link to the resulting swagger hub page from our documentation.

## Important notes
- We need to create an EDC account with Swaggerhub. @mspiekermann is that something you could help out with?
- Once we have an always-on MVD deployment, we could have the SwaggerHub page point to it in its `server` property, i.e. users can try out the the DataManagement API live.

## Which Areas Would Be Affected?
Documentation, CI workflows

## Why Is the Feature Desired?
Documentation, visually inspecting the API.

## Solution Proposal
- add a CI job that publishes the `openapi.yaml` to SwaggerHub
- add a link to our documentation pointing to the generated website

## Type of Issue
improvement

### PR DESCRIPTION
**Pull request by paullatzelsperger**
## What this PR changes/adds

Adds a decision record about how we want to refactor and improve our APIs and its documentation.

## Why it does that

Make decisions traceable.

## Further notes

- this DR consists of two files, one dealing with the renaming side of things, the other with how the documentation process will be refactored.

## Linked Issue(s)

Related #2163 

## Checklist

- [ ] added appropriate tests?
- [ ] performed checkstyle check locally?
- [ ] added/updated copyright headers?
- [ ] documented public classes/methods?
- [ ] added/updated relevant documentation?
- [ ] assigned appropriate label? (exclude from changelog with label `no-changelog`)
- [ ] formatted title correctly? (_take a look at the [CONTRIBUTING](https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/blob/main/CONTRIBUTING.md#submit-a-pull-request) and [Etiquette for pull requests](https://github.com/eclipse-dataspaceconnector/DataSpaceConnector/blob/main/pr_etiquette.md) for details_)

### CODE REVIEW

**Comment by paullatzelsperger** [comment]
@ndr-brt since you raised the issue about renaming the APIs, I'd appreciate your feedback and additional input, esp. on the renaming part of this DR.

**Review by ndr-brt — APPROVED** [review]

File: docs/developer/decision-records/README.md [hunk 1]
```diff
@@ -43,3 +43,4 @@
 - [2022-10-20 Trust Framework Adoption repository](2022-10-20-trust-framework-adoption-repository)
 - [2022-10-21 Gradle Version Catalogs](2022-10-21-gradle-versioncatalogs/)
 - [2022-10-31 Service Layer](2022-10-31-aggregate-service-layer)
```
Comment (juliapampus) [inline docs/developer/decision-records/README.md:45]:
```suggestion
- [2022-10-31 Service Layer](2022-10-31-aggregate-service-layer/)
```

File: docs/developer/decision-records/README.md [hunk 2]
```diff
@@ -43,3 +43,4 @@
...
+- [2022-11-09 API Restructuring](2022-11-09-api-refactoring)
```
Comment (juliapampus) [inline docs/developer/decision-records/README.md:46]: This does not work as the path/ automatically points to a `README.md`. Pls refer to a document and within this provide a link to the other one. Or create a README.md that lists the links to both files.

**Review by juliapampus — CHANGES_REQUESTED** [review]
I agree with the content in general :)

File: docs/developer/decision-records/README.md [hunk 3]
```diff
@@ -43,3 +43,4 @@
 - [2022-10-20 Trust Framework Adoption repository](2022-10-20-trust-framework-adoption-repository)
```
Comment (juliapampus) [inline docs/developer/decision-records/README.md:43]:
```suggestion
- [2022-10-20 Trust Framework Adoption repository](2022-10-20-trust-framework-adoption-repository/)
```

Comment (juliapampus) [reply docs/developer/decision-records/README.md:43]: (Just because I came across this now)

File: docs/developer/decision-records/README.md (hunk 2 above)
Comment (paullatzelsperger) [reply docs/developer/decision-records/README.md:46]: good to know!

**Pull request merged**